clouddriver-aws/src/main/java/com/netflix/spinnaker/clouddriver/aws/lifecycle/InstanceTerminationLifecycleWorker.java

```diff
@@ -227,24 +218,31 @@ private boolean updateInstanceStatus(Eureka eureka, String app, String instanceI
       try {
         eureka.updateInstanceStatus(app, instanceId, DiscoveryStatus.OUT_OF_SERVICE.getValue());
         return true;
-      } catch (RetrofitError e) {
-        final String recoverableMessage =
-            "Failed marking app out of service (status: {}, app: {}, instance: {}, retry: {})";
-        if (HttpStatus.NOT_FOUND.value() == e.getResponse().getStatus()) {
-          log.warn(recoverableMessage, e.getResponse().getStatus(), app, instanceId, retry);
-        } else if (e.getKind() == RetrofitError.Kind.NETWORK) {
-          log.error(recoverableMessage, e.getResponse().getStatus(), app, instanceId, retry, e);
-        } else {
-          log.error(
-              "Irrecoverable error while marking app out of service (app: {}, instance: {}, retry: {})",
-              app,
-              instanceId,
-              retry,
-              e);
-          break;
-        }
+      } catch (NotFoundException e) {
```

Comment:
SpinnakerHttpException with a 404 response code is the equivalent of RetrofitError with a 404 response code -- not NotFoundException.